payment/src/main/java/org/killbill/billing/payment/api/DefaultApiBase.java

```diff
@@ -79,6 +80,13 @@ protected void logAPICall(final String transactionType, final Account account, f
                 logLine.append(", currency = ")
                        .append(currency);
             }
+
+            final String requestId = Request.getPerThreadRequestData() != null
```

Comment:
I am not sure if I understand it correctly. How is the userToken normally filled? Is that something that happens completely on behalf of the client? So, Groupon could for example decide to put the request id into userToken? I am just wondering how we enable them to analyze cross application requests in the future, because I convinced that they don't want to lose this ability.